sections/dir.zsh

```diff
@@ -21,14 +22,34 @@ SPACESHIP_DIR_COLOR="${SPACESHIP_DIR_COLOR="cyan"}"
 spaceship_dir() {
   [[ $SPACESHIP_DIR_SHOW == false ]] && return
 
-  local dir
+  local dir trunc_prefix
 
   # Threat repo root as a top-level directory or not
   if [[ $SPACESHIP_DIR_TRUNC_REPO == true ]] && spaceship::is_git; then
     local git_root=$(git rev-parse --show-toplevel)
-    dir="$git_root:t${$(expr $(pwd) : "$git_root\(.*\)")}"
+
+    # Check if the parent of the $git_root is "/"
+    if [[ $git_root:h == / ]]; then
+      trunc_prefix=/
+    else
+      trunc_prefix=$SPACESHIP_DIR_TRUNC_PREFIX
+    fi
+
+    # `${NAME#PATTERN}` removes a leading prefix PATTERN from NAME.
+    # `$~~` avoids `GLOB_SUBST` so that `$git_root` won't actually be
+    # considered a pattern and matched literally, even if someone turns that on.
+    # See "Parameter Expansion" under the Zsh manual.
+    dir="$trunc_prefix$git_root:t${PWD#$~~git_root}"
```

Comment:
This causes symlinked directories to appear twice.

Reply:
This is because `$git_root` has symlinks resolved. Guess I can use the working directory with symlinks resolved here. Hopefully that's the desired behavior.